src/selectors/linked-nodes.js

```diff
@@ -17,36 +15,83 @@ export const getCentralNode = createSelector(
 );
 
 /**
- * Recursively search through the edges data for ancestor and descendant nodes
+ * Gets a map of nodeIDs to the visible source edges and target edges
  * @param {Array} edges
+ */
+export const getVisibleEdgesByNode = createSelector(
+  [getVisibleEdges],
+  edges => {
+    const sourceEdges = {};
+    const targetEdges = {};
+
+    for (const edge of edges) {
+      if (!sourceEdges[edge.target]) {
+        sourceEdges[edge.target] = [];
+      }
+
+      sourceEdges[edge.target].push(edge);
+
+      if (!targetEdges[edge.source]) {
+        targetEdges[edge.source] = [];
+      }
+
+      targetEdges[edge.source].push(edge);
+    }
+
+    return { sourceEdges, targetEdges };
+  }
+);
+
+/**
+ * Finds all visible successors for the given nodeID in given direction
+ * @param {string} nodeID
+ * @param {Object} edgesByNode
+ * @param {string} direction
+ * @param {?object} visited
+ * @param {?boolean} isStart
+ * @param {?Array} empty
```

Comment:
Can you please add descriptions for each of these arguments?

Reply:
Sure I've added some info here